### PR TITLE
Vector.rotate, and fix for stopListening

### PR DESCRIPTION
**Pull request by ferm10n**
I found a problem with my earlier implementation of listenTo. Since event handlers are wrapped in a function with a different scope, when calling .off() the specified callback won't match the callback in _events (since its wrapped in another function created by listenTo). There's probably a better way to fix this than the solution I have here, but for now it works!

Also, I added a helpful function for rotating individual vectors!

### CODE REVIEW

File: src/vector.js [hunk 1]
```diff
@@ -141,6 +141,13 @@
 
     toObject: function() {
       return { x: this.x, y: this.y };
+    },
+
+    rotate: function (radians) {
+      var newX = this.x * Math.cos(radians) - this.y * Math.sin(radians);
+      var newY = this.x * Math.sin(radians) + this.y * Math.cos(radians);
+      this.set(newX, newY);
```
Comment (GoToLoop) [inline src/vector.js]: Method **set()** is already chainable. So we can use `return this.set(newX, newY);` right way.

File: src/vector.js [hunk 2]
```diff
@@ -141,6 +141,12 @@
 
     toObject: function() {
       return { x: this.x, y: this.y };
+    },
+
+    rotate: function (radians) {
+      var newX = this.x * Math.cos(radians) - this.y * Math.sin(radians);
+      var newY = this.x * Math.sin(radians) + this.y * Math.cos(radians);
+      return this.set(newX, newY);;
```
Comment (GoToLoop) [inline src/vector.js]: Double semi-colon here: `;;`

**Comment by ferm10n** [comment]
\*facepalm\*
@GoToLoop, fixed err 754

See https://github.com/joho/7XX-rfc ;)

**Comment by GoToLoop** [comment]
Oh, I've got another suggestion. There are double **Math.cos()** & **Math.sin()** calculations!
Seems like a good idea to cache them both for performance reasons.What do you think?
```js
    rotate: function (radians) {
      var cos = Math.cos(radians),
          sin = Math.sin(radians),
          newX = this.x*cos - this.y*sin,
          newY = this.x*sin + this.y*cos;
      return this.set(newX, newY);
    }
```

**Comment by jonobr1** [comment]
Looks fine to me. We should consider writing some tests for the new items you added -- this way we'll know in the future if something works correctly or not 👍 

Also, is `listenTo` and `stopListening` method names from an existing library? Two.js uses `listen` and `ignore` in `Two.Anchor`. I think it would be more consistent to use that nomenclature if there isn't a precedent elsewhere.

**Comment by ferm10n** [comment]
Agreed!

I got the names listenTo and stopListening from Backbone's event thing. I don't see any reason not to change it.

**Comment by jonobr1** [comment]
All righty, this is `dev` and I updated with `listenTo` -> `listen` and `stopListening` to `ignore`. Thanks for the PR!

**Pull request merged**